internal/employees/repository/EmployeeRepository.go

```diff
@@ -1,77 +1,99 @@
 }
 
-func (repository) cardNumberIsUnique(cardNumberId string) bool {
-	for _, employee := range employees {
-		if employee.CardNumberId == cardNumberId {
-			return false
+func (repo *mariaDBEmployeerepository) GetAll(ctx context.Context) ([]domain.Employee, error) {
+	employees := []domain.Employee{}
+
+	rows, err := repo.db.QueryContext(ctx, SQLFindAllEmployees)
+
+	if err != nil {
+		return employees, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var employee domain.Employee
+		if err := rows.Scan(&employee.Id, &employee.CardNumberId, &employee.FirstName, &employee.LastName, &employee.WarehouseId); err != nil {
+			return nil, err
 		}
+		employees = append(employees, employee)
 	}
-	return true
-}
 
-func (repository) GetAll() ([]domain.Employee, error) {
 	return employees, nil
 }
 
-func (repository) GetById(id int64) (*domain.Employee, error) {
-	for _, employee := range employees {
-		if employee.Id == id {
-			return &employee, nil
-		}
+func (repo *mariaDBEmployeerepository) GetById(ctx context.Context, id int64) (*domain.Employee, error) {
+	var employee domain.Employee
+
+	row := repo.db.QueryRowContext(ctx, SQLFindEmployeeByID, id)
+	err := row.Scan(&employee.Id, &employee.CardNumberId, &employee.FirstName, &employee.LastName, &employee.WarehouseId)
+	if err != nil {
+		return nil, domain.ErrEmployeeNotFound
 	}
 
-	return nil, domain.ErrEmployeeNotFound
+	return &employee, nil
 }
 
-func (repo repository) Create(cardNumberId string, firstName string, lastName string, warehouseId int64) (domain.Employee, error) {
-	nextId := lastId
+func (repo *mariaDBEmployeerepository) Create(ctx context.Context, cardNumberId string, firstName string, lastName string, warehouseId int64) (domain.Employee, error) {
 	employee := domain.Employee{
-		Id:           nextId,
 		CardNumberId: cardNumberId,
 		FirstName:    firstName,
 		LastName:     lastName,
 		WarehouseId:  warehouseId,
 	}
-
-	if !repo.cardNumberIsUnique(cardNumberId) {
-		return domain.Employee{}, domain.ErrCardNumberMustBeUnique
+	res, err := repo.db.ExecContext(
+		ctx,
+		SQLCreateEmployee,
+		&employee.CardNumberId, &employee.FirstName, &employee.LastName, &employee.WarehouseId,
+	)
+	if err != nil {
+		return domain.Employee{}, err
 	}
 
-	employees = append(employees, employee)
-	lastId += 1
+	id, _ := res.LastInsertId()
+	employee.Id = int64(id)
+
 	return employee, nil
 }
 
-func (repo repository) UpdateFullname(id int64, firstName string, lastName string) (*domain.Employee, error) {
-	for i, employee := range employees {
-		if employee.Id == id {
-			employees[i].FirstName = firstName
-			employees[i].LastName = lastName
-			return &employees[i], nil
-		}
+func (repo mariaDBEmployeerepository) Update(ctx context.Context, employeeID int64, updatedEmployee domain.Employee) error {
+	_, err := repo.db.ExecContext(
+		ctx,
+		SQLUpdateEmployeeFullname,
+		updatedEmployee.FirstName, updatedEmployee.LastName, employeeID,
+	)
+
+	if err != nil {
+		return err
```

Comment:
Faltou tratar aqui o:
	rowsAffected, _ := result.RowsAffected()
	if rowsAffected == 0 {
		return nil, errors.New("employe not found")
	}